packages/react-api/src/api/SQL.js

```diff
@@ -41,16 +50,23 @@ export const executeSQL = async (credentials, query, opts = { format: '' }) => {
  * Create an 'SQL query' request
  * (using GET or POST request, depending on url size)
  */
-function createRequest({ credentials, query, opts }) {
+function createRequest({ credentials, connection, query, opts = {} }) {
   const { abortController, ...otherOptions } = opts;
 
+  const { apiVersion = API_VERSIONS.V2 } = credentials;
+
   const rawParams = {
-    api_key: credentials.apiKey,
-    client: credentials.username,
-    q: query.trim(),
+    client: 'carto-react',
+    q: query?.trim(),
     ...otherOptions
   };
 
+  if (apiVersion === API_VERSIONS.V3) {
+    rawParams.access_token = credentials.accessToken;
```

Comment:
It looks like the token we're passing here will appear in the url, is that fine? Another option would be trying to inject it in the headers, via requestOpts...

Reply:
Yes, I do that in deck.gl module. https://github.com/visgl/deck.gl/blob/master/modules/carto/src/api/maps-cloud-native-client.js#L21

I didn't do here because the purpose of doing that is to skip the cache, but sql endpoint is not cached in cloud native api

Reply:
Maybe I can do it, but I prefer in a separated PR as an improvement

Reply:
Cool